src/site/topics/objects-review/equals.rst

```diff
@@ -1,56 +1,167 @@
+
+* Lastly, it check if all the attributes are equal
+
+.. note::
+
+    It is important to understand the difference between ``someObject == someOtherObject``,
+    ``someObject.equals(someOtherObject)``, and ``Objects.equals(someObject, someOtherObject)``.
+
+    With ``someObject == someOtherObject``, it checks if ``someObject`` and ``someOtherObject`` are referencing the same
+    object -- aliases.
+
+    ``someObject.equals(someOtherObject)`` checks if ``someObject`` and ``someOtherObject`` are equivalent based on
+    ``someObject`` class' ``equals`` method.
+
+    ``Objects.equals(someObject, someOtherObject)`` also checks if ``someObject`` is ``null``, and then checks if
+    ``someObject`` and ``someOtherObject`` are equivalent based on ``someObject`` class' ``equals`` method.
+
+    The below example makes use of the third option to be safe around ``null``, but realistically, based on the way the
+    ``Friend`` class is written, it is not possible for an instance of a ``Friend`` object to be created with ``null``
+    fields. In other words, the second option would be fine, but using the third option would still be more defensive.
+
+
+.. code-block:: java
+    :linenos:
+
+    /**
+     * Checks if two Friend objects are equal. Friend objects are considered equal if all their attributes are equal.
+     *
+     * @param o an "object" being compared to.
+     * @return True if the two objects are equal, false otherwise.
+     */
+    public boolean equals(Object o) {
+        // If o is actually in the same memory address of this
+        if (o == this) {
+            return true;
+        }
+        // If o is null, then it's not equal
+        if (o == null) {
+            return false;
+        }
+        // if o and this are of different classes, they're not the equal
+        if (o.getClass() != this.getClass()) {
+            return false;
+        }
+        // Cast o as a friend
+        Friend other = (Friend) o;
+        return Objects.equals(this.firstName, other.firstName) &&
+                Objects.equals(this.lastName, other.lastName) &&
+                Objects.equals(this.email, other.email);
+    }
+
+
+* Below is an example of the ``equals`` method for the ``ContactList`` class
+
+    * Notice that it follows the same pattern
+    * The only difference is the check on what needs to be equal
+    * Here, ``Arrays.equals`` is used to check the equality on the array
+
+        * The alternative would be to loop over the array and check equality on each element within the loop
+
+
+.. code-block:: java
+    :linenos:
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        ContactList that = (ContactList) o;
+        return this.size == that.size && Arrays.equals(this.friends, 0, this.size(), that.friends, 0, that.size());
```

Comment:
The `size` comparison is unnecessary, `Arrays.equals` will do that.

Reply:
:+1: